alf/utils/lean_function.py

```diff
@@ -13,67 +13,123 @@
 # limitations under the License.
 
 import torch
+import torch.nn as nn
+import types
+from typing import Callable
+
+from alf.nest import flatten, pack_sequence_as
+from alf.networks import Network
 
 
 class _LeanFunction(torch.autograd.Function):
+    # Reference: Pytorch: Defining new Autograd Functions
+    # https://pytorch.org/tutorials/beginner/examples_autograd/two_layer_net_custom_function.html
     @staticmethod
-    def forward(ctx, func, keywords, *args):
+    def forward(ctx, func, num_parameters, keywords, *args):
+        """
+        Args:
+            ctx (_LeanFunction):
+            func (Callable): func/module to be wrapped
+            num_parameters (int): the number of nn.Parameters of func if it is an
```

Comment:
I know this is not public facing documentation, but it would help me if we explain the pattern for `forward`: for modules, the first `num_parameters` arguments in `args` are the parameters of that module.

Reply:
added.